From: "You-Sheng Yang (vicamo)" <vicamo.yang@canonical.com>
Date: Fri, 29 Apr 2022 15:20:02 +0800
Subject: compat: Remove usage of the deprecated "pci-dma-compat.h" API

Follow commit 7968778914e5 ("PCI: Remove the deprecated
"pci-dma-compat.h" API").

Signed-off-by: You-Sheng Yang (vicamo) <vicamo.yang@canonical.com>
---
 drivers/media/pci/intel/ipu.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/media/pci/intel/ipu.c b/drivers/media/pci/intel/ipu.c
index 2cadafc..063f7b9 100644
--- a/drivers/media/pci/intel/ipu.c
+++ b/drivers/media/pci/intel/ipu.c
@@ -440,10 +440,10 @@ static int ipu_pci_probe(struct pci_dev *pdev, const struct pci_device_id *id)
 	dev_dbg(&pdev->dev, "isys_base: 0x%lx\n", (unsigned long)isys_base);
 	dev_dbg(&pdev->dev, "psys_base: 0x%lx\n", (unsigned long)psys_base);
 
-	rval = pci_set_dma_mask(pdev, DMA_BIT_MASK(dma_mask));
+	rval = dma_set_mask(&pdev->dev, DMA_BIT_MASK(dma_mask));
 	if (!rval)
-		rval = pci_set_consistent_dma_mask(pdev,
-						   DMA_BIT_MASK(dma_mask));
+		rval = dma_set_coherent_mask(&pdev->dev,
+					     DMA_BIT_MASK(dma_mask));
 	if (rval) {
 		dev_err(&pdev->dev, "Failed to set DMA mask (%d)\n", rval);
 		return rval;
